app/controllers/AnnotationPrivateLinkController.scala

```diff
@@ -0,0 +1,102 @@
+package controllers
+
+import com.mohiva.play.silhouette.api.Silhouette
+import com.scalableminds.util.accesscontext.GlobalAccessContext
+import com.scalableminds.util.tools.Fox
+import com.scalableminds.util.tools.FoxImplicits
+import io.swagger.annotations._
+import play.api.libs.json._
+
+import javax.inject.Inject
+import models.annotation._
+import oxalis.security.WkEnv
+import play.api.mvc.{Action, AnyContent, PlayBodyParsers}
+import utils.ObjectId
+
+import scala.concurrent.ExecutionContext
+@Api
+class AnnotationPrivateLinkController @Inject()(
+    annotationDAO: AnnotationDAO,
+    annotationService: AnnotationService,
+    annotationPrivateLinkDAO: AnnotationPrivateLinkDAO,
+    annotationPrivateLinkService: AnnotationPrivateLinkService,
+    sil: Silhouette[WkEnv])(implicit ec: ExecutionContext, val bodyParsers: PlayBodyParsers)
+    extends Controller
+    with FoxImplicits {
+
+  @ApiOperation(hidden = true, value = "")
+  def lookupPrivateLink(accessToken: String): Action[AnyContent] = Action.async { implicit request =>
+    for {
+      annotationPrivateLink <- annotationPrivateLinkDAO.findOneByAccessToken(accessToken)
+      _ <- bool2Fox(annotationPrivateLink.expirationDateTime.forall(_ > System.currentTimeMillis())) ?~> "Token expired" ~> 404
+      annotation: Annotation <- annotationDAO.findOne(annotationPrivateLink._annotation)(GlobalAccessContext)
+      writtenAnnotation <- annotationService.writesAsAnnotationSource(annotation)
+    } yield Ok(writtenAnnotation)
+  }
+
+  def list: Action[AnyContent] = sil.SecuredAction.async { implicit request =>
+    for {
+      links <- annotationPrivateLinkDAO.findAll
+      linksJsonList <- Fox.serialCombined(links)(annotationPrivateLinkService.publicWrites)
+    } yield Ok(Json.toJson(linksJsonList))
+  }
+
+  def listByAnnotation(annotationId: String): Action[AnyContent] = sil.SecuredAction.async { implicit request =>
+    for {
+      annotationIdValidated <- ObjectId.fromString(annotationId)
+      links <- annotationPrivateLinkDAO.findAllByAnnotation(annotationIdValidated)
+      linksJsonList <- Fox.serialCombined(links)(annotationPrivateLinkService.publicWrites)
+    } yield Ok(Json.toJson(linksJsonList))
+  }
+
+  def get(id: String): Action[AnyContent] = sil.SecuredAction.async { implicit request =>
+    for {
+      idValidated <- ObjectId.fromString(id)
+
+      annotationPrivateLink <- annotationPrivateLinkDAO.findOne(idValidated)
+      _ <- bool2Fox(annotationPrivateLink.expirationDateTime.forall(_ > System.currentTimeMillis())) ?~> "Token expired" ~> 404
+      _ <- annotationDAO.findOne(annotationPrivateLink._annotation) ?~> "annotation.notFound" ~> NOT_FOUND
+
+      annotationPrivateLinkJs <- annotationPrivateLinkService.publicWrites(annotationPrivateLink)
+    } yield Ok(annotationPrivateLinkJs)
+  }
+
+  def create: Action[AnnotationPrivateLinkParams] = sil.SecuredAction.async(validateJson[AnnotationPrivateLinkParams]) {
```

Comment:
@leowe Please add some more information to the swagger annotations. All of those endpoints should have a proper `operationId` in https://zarrprivatelinks.webknossos.xyz/swagger.json, by default it is the function name. You can specify a different one via `nickname`, as seen on some other endpoints. Please also add some descriptions, especially for endpoints that expect a specific json format.

Reply:
Only for the wk endpoints should be enough :+1: Let's see what we come up with tomorrow.

Reply:
Thanks!